src/Moq/MatcherFactory.cs

```diff
@@ -132,9 +132,13 @@ internal static class MatcherFactory
 				}
 #pragma warning restore 618
 
-				return new Pair<IMatcher, Expression>(new LazyEvalMatcher(originalExpression), originalExpression);
+				var method = call.Method;
+				if (!method.IsPropertyGetter() && !method.IsPropertyIndexerGetter())
```

Comment:
This actually does a little more than target indexers. It also makes sure that if somehow a property or index expression has made it into the setup expression in the form of a `get_*` method call, we recognize them anyway.